api/include/opentelemetry/trace/span_context_kv_iterable.h

```diff
@@ -30,5 +30,22 @@ class SpanContextKeyValueIterable
    */
   virtual size_t size() const noexcept = 0;
 };
+
+/**
+ * @brief Null Span context that does not carry any information.
+ */
+class NullSpanContext : public SpanContextKeyValueIterable
```

Comment:
It's a long name, but let's use `NullSpanContextKeyValueIterable`. To be consistent with `SpanContextKeyValueIterable`. `NullSpanContext` is ambiguous.

Reply:
Anyway, to avoid an additional class, one could also do this:
```c++
using SpanContextKeyValueIterableInitializerList = SpanContextKeyValueIterableView<std::initializer_list<
          std::pair<SpanContext,
                    std::initializer_list<std::pair<nostd::string_view, common::AttributeValue>>>>>;
```

And then define your additional `StartSpan` method like this:
```c++
 nostd::shared_ptr<Span> StartSpan(nostd::string_view name,
                                    const common::KeyValueIterable &attributes,
                                    const StartSpanOptions &options = {}) noexcept
  {
    return this->StartSpan(name, attributes, SpanContextKeyValueIterableInitializerList({}), options);
  }
```

Just to avoid a clutter of types that we also need to document (and that are error-prone, e. g. return `false` in case of success). We also have `NullKeyValueIterable`, however, it's not used anywhere ...

Reply:
I can refactor this in a separate (unrelated PR) by moving these into separate helper header or something. That would also enable `NullTracerProvider` cases, unrelated to ETW per se. Right now it is more or less "internal" detail that I think we can rename / adjust before GA.